ios/RNShopify.m

```diff
@@ -0,0 +1,320 @@
+
+#import "RNShopify.h"
+#import "Buy.h"
+
+@implementation RNShopify {
+    RCTPromiseResolveBlock _resolve;
+    RCTPromiseRejectBlock _reject;
+}
+
+- (dispatch_queue_t)methodQueue
+{
+    return dispatch_get_main_queue();
+}
+
+RCT_EXPORT_MODULE()
+
+RCT_EXPORT_METHOD(initialize:(NSString *)domain key:(NSString *)key)
+{
+    //Application ID is always 8, as stated in official documentation from Shopify
+    self.client = [[BUYClient alloc] initWithShopDomain:domain
+                                                 apiKey:key
+                                                  appId:@"8"];
+}
+
+RCT_EXPORT_METHOD(getShop:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getShop:^(BUYShop *shop, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve([shop JSONDictionary]);
+    }];
+}
+
+RCT_EXPORT_METHOD(getCollections:(NSUInteger)page resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getCollectionsPage:page completion:^(NSArray<BUYCollection *> *collections, NSUInteger page, BOOL reachedEnd, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        NSMutableArray *collectionDictionaries = [NSMutableArray array];
+
+        for (BUYCollection *collection in collections) {
+            [collectionDictionaries addObject: @{@"title":collection.title,@"id":collection.identifier}];
```

Comment:
How are identifiers converted to JSON in other cases? I see that you are mapping it to `id` here, is this consistent with other objects? Also, extract this as a new function, add a comment that this is here because of a bug, and include a link to this bug.